src/pages/IdeLoader/index.tsx

```diff
@@ -0,0 +1,225 @@
+  PageSection, PageSectionVariants, Tab, Tabs, Wizard,
+  WizardStep,
+} from '@patternfly/react-core';
+import { CheckCircleIcon } from '@patternfly/react-icons';
+import { ExclamationCircleIcon, InProgressIcon } from '@patternfly/react-icons/dist/js/icons';
+import React, { RefObject } from 'react';
+import Header from '../../components/Header';
+import LogsTab from '../../components/LogsTab';
+import { WorkspaceStatus } from '../../services/workspaceStatus';
+
+import styles from '../../components/WorkspaceStatusLabel/WorkspaceStatusLabel.module.css';
+import './index.styl';
+
+export const SECTION_THEME = PageSectionVariants.light;
+
+export enum IdeLoaderTabs {
+  Progress = 0,
+  Logs = 1,
+}
+
+type Props = {
+  hasError: boolean,
+  currentStep: number,
+  workspaceName: string;
+  workspaceId: string;
+  ideUrl?: string;
+  callbacks?: {
+    showAlert?: (variant: AlertVariant.success | AlertVariant.danger, title: string) => void
+  }
+};
+
+type State = {
+  alertVisible?: boolean;
+  activeTabKey?: IdeLoaderTabs;
+  stepIdReached?: number;
+  currentRequestError?: string;
+};
+
+class IdeLoader extends React.PureComponent<Props, State> {
+  private alert: { variant?: AlertVariant.success | AlertVariant.danger; title?: string } = {};
+  public showAlert: (variant: AlertVariant.success | AlertVariant.danger, title: string, timeDelay?: number) => void;
+  private readonly hideAlert: () => void;
+  private readonly handleTabClick: (event: any, tabIndex: any) => void;
+
+  private readonly wizardRef: RefObject<any>;
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      alertVisible: false,
+      activeTabKey: IdeLoaderTabs.Progress,
+      stepIdReached: 1,
+      currentRequestError: '',
+    };
+
+    this.wizardRef = React.createRef();
+
+    // Toggle currently active tab
+    this.handleTabClick = (event: any, tabIndex: any): void => {
+      this.setState({ activeTabKey: tabIndex });
+      if (this.state.activeTabKey === IdeLoaderTabs.Progress) {
+        this.setState({ alertVisible: false });
+      }
+    };
+    // Init showAlert
+    let showAlertTimer;
+    this.showAlert = (variant: AlertVariant.success | AlertVariant.danger, title: string): void => {
+      this.setState({ currentRequestError: title });
+      if (this.state.activeTabKey === IdeLoaderTabs.Progress) {
+        return;
+      }
+      this.alert = { variant, title };
+      this.setState({ alertVisible: true });
+      if (showAlertTimer) {
+        clearTimeout(showAlertTimer);
+      }
+      showAlertTimer = setTimeout(() => {
+        this.setState({ alertVisible: false });
+      }, variant === AlertVariant.success ? 2000 : 10000);
+    };
+    this.hideAlert = (): void => this.setState({ alertVisible: false });
+    // Prepare showAlert as a callback
+    if (this.props.callbacks && !this.props.callbacks.showAlert) {
+      this.props.callbacks.showAlert = (variant: AlertVariant.success | AlertVariant.danger, title: string) => {
```

Comment:
```suggestion
      this.props.callbacks.showAlert = (variant: AlertVariant, title: string) => {
```

Reply:
Agreed to extend types for the future. I have fixed it.